reactioncommerce/reaction
# refactor: remove all grid positions code and UI

Resolves #4542   
Impact: **minor**  
Type: **refactor**

## Issue
The customer-facing grid UI no longer supports manually positioned or pinned items, or differing weights. Features similar to this will be reimplemented in a different way. However, the operator UI for the grid still allowed you drag and drop items and change their weights.

## Solution
When logged in as a user with "createProduct" permission and with edit mode on, you can no longer drag and drop items on the grid and the "Size" area in the panel no longer appears. All grid items are always "small" size and in ascending order by creation date (for now).

Pulling this code out caused the grid to have rerendering issues, which were fixed by some refactoring of the grid item selection code to make it cleaner and less prone to excessive rerendering.

## Testing
1. Log in as a user with "createProduct" permission.
2. Toggle on edit mode in the upper left corner.
3. Try to drag/drop items on the grid and verify that you cannot.
4. Click items in the grid. Verify they still appear in the action panel, but that the "Size" area allowing you to choose different "weights" for the item is no longer there.
